Make the `MovingWindow` public and remove public mentions to the `RingBuffer`

leandro-lucarella-frequenz opened the pull request:
The ring buffer is not part of the public interface yet, so it shouldn't be shown in the release notes of the user facing documentation.

The `MovingWindow` is now exported in the `frequenz.sdk.timeseries` package. The `ResamplerConfig` class is also now made public via the same package, as it is part of the `MovingWindow`'s public interface.

llucax committed:
Don't make public mentions to the RingBuffer

The ring buffer is not part of the public interface yet, so it shouldn't
be shown in the release notes of the user facing documentation.

The bug in the ring buffer potentially affecting the MovingWindow  is
reworded to include the MovingWindow.

Signed-off-by: Leandro Lucarella <leandro.lucarella@frequenz.com>

llucax committed:
Make the MovingWindow class public

The MovingWindow is now exported in the frequenz.sdk.timeseries package.
The ResamplerConfig class is also now made public via the same package,
as it is part of the MovingWindow's public interface.

Signed-off-by: Leandro Lucarella <leandro.lucarella@frequenz.com>

daniel-zullo-frequenz commented on src/frequenz/sdk/timeseries/_moving_window.py:
```diff
@@ -36,7 +36,7 @@ class MovingWindow:
     Note that a numpy ndarray is returned and thus users can use
     numpys operations directly on a window.
 
-    The window uses an ringbuffer for storage and the first element is aligned to
```
Optionally you might want to add a new commit to add the changes related to fixing documentation in  MovingWindow

leandro-lucarella-frequenz replied:
I think it is related to considering the ring buffer implementation as internal, so I put them together on purpose.

daniel-zullo-frequenz replied:
ok, then resolving it...

daniel-zullo-frequenz commented:
I have only an optional suggestion, though pylint is not happy


************* Module noxfile
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._moving_window) (cyclic-import)
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._moving_window -> src.frequenz.sdk.timeseries._resampling) (cyclic-import)
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._resampling) (cyclic-import)
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._moving_window -> src.frequenz.sdk.timeseries._ringbuffer -> src.frequenz.sdk.timeseries._ringbuffer.serialization -> src.frequenz.sdk.timeseries._ringbuffer.buffer) (cyclic-import)
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._moving_window -> src.frequenz.sdk.timeseries._ringbuffer -> src.frequenz.sdk.timeseries._ringbuffer.buffer) (cyclic-import)

leandro-lucarella-frequenz commented:
Damn, this is why we didn't put many symbols in the module. It is really annoying. 🙄

llucax committed:
Fix some imports to avoid cycles

Import should always use the leaf modules, as importing modules from the
package usually leads to cyclic imports.

Signed-off-by: Leandro Lucarella <leandro.lucarella@frequenz.com>

leandro-lucarella-frequenz commented:
Update to fix the cyclic imports.

FYI @matthias-wende-frequenz and @mathias-baumann-frequenz (have a look at the last commit, remember to always import symbols from leaf modules instead of packages).
